### PR TITLE
Feature/dev 4213 circuit breaker

### PR DESCRIPTION
**Pull request by wolff95**
<!--

# 🎉 Thanks for taking the time to contribute to Findify! 🎉

It is highly appreciated that you take the time to help improve Findify.
We appreciate it if you would take the time to document your Pull Request.

Before submitting a new PR please read the [CONTRIBUTING.md](./CONTRIBUTING.md).

### CODE REVIEW

File: packages/sdk/src/utils.ts [hunk 1]
```diff
@@ -5,21 +5,64 @@ export const omit = <TR>(ks: string[], original: object) =>
...
+ * 
+ * @param key widgetId + MerchantId - key of the session storage value
+ * @returns failureCount and lastFailureDate of the matched key
+ */
+const getCircuitBreakerStatus = (key: string): { failureCount: number; lastFailureDate: Date; } => {
+  const findifyCircuitBreaker = sessionStorage.getItem(key);
+  return findifyCircuitBreaker ? JSON.parse(findifyCircuitBreaker) : { failureCount: null, lastFailureDate: null };
+}
+
+const tryAgainCheck = (lastFailure) => {
```
Comment (pascalachkar) [inline packages/sdk/src/utils.ts:39]: Let's add parameter type

File: packages/sdk/src/utils.ts [hunk 2]
```diff
@@ -5,21 +5,64 @@ export const omit = <TR>(ks: string[], original: object) =>
...
+  const lastAttempt = new Date(lastFailure);
```
Comment (pascalachkar) [inline packages/sdk/src/utils.ts:40]: Handle error if variable is not date, this would throw an error

File: packages/sdk/src/utils.ts [hunk 3]
```diff
@@ -5,21 +5,64 @@ export const omit = <TR>(ks: string[], original: object) =>
...
+/**
+ * 
+ * @param fn Request to execute
+ * @param key ${merchantId}_${widgetId} for session storage
+ * @returns 
+ */
+export const requestHandler = (fn: RetryCallback, key) =>
+  new Promise((resolve, reject) => {
+    const { failureCount, lastFailureDate } = getCircuitBreakerStatus(key);
+    if (blockRequest(failureCount, lastFailureDate)) return reject('Too many attemps, please try again');
```
Comment (pascalachkar) [inline packages/sdk/src/utils.ts:61]: Typo 'attempts' 
Maybe we can change the message a bit, as most merchants will see it, 'Too many attempts, please retry after a few minutes' or smtg like that wdyt @wolff95

File: packages/sdk/src/utils.ts (hunk 2 above)
Comment (wolff95) [reply packages/sdk/src/utils.ts:40]: Just added a lastAttempty?.getTime() so if it is invalid at least there is no error.
I do not expect this to not be a date

**Review by pascalachkar — APPROVED** [review]
Approved !!

**Pull request merged**